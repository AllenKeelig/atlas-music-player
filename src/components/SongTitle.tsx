import { ReactNode } from "react";

type SongTitleProps = {
  artistName: string;
  children: ReactNode;
};

const SongTitle: React.FC<SongTitleProps> = ({ artistName, children }) => {
  return (
    <div className="text-center">
      <h2 className="mb-2 text-2xl font-bold">{children}</h2>
      <p className="text-sm text-accent">{artistName}</p>
    </div>
  );
};

export default SongTitle;
