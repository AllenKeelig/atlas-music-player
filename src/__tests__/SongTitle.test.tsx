// src/__tests__/SongTitle.test.tsx
import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import SongTitle from "../components/SongTitle";

test("SongTitle renders with title and artist", () => {
  const { container } = render(
    <SongTitle title="Cool Song" artist="Cool Artist" />
  );
  expect(container).toMatchSnapshot();
});

test("SongTitle renders with only title", () => {
  const { container } = render(<SongTitle title="Lonely Song" />);
  expect(container).toMatchSnapshot();
});
