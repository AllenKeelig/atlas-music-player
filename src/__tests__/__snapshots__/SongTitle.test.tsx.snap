// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle renders with only title 1`] = `
<div>
  <div
    class="text-center"
  >
    <h2
      class="mb-2 text-2xl font-bold"
    />
    <p
      class="text-sm text-accent"
    />
  </div>
</div>
`;

exports[`SongTitle renders with title and artist 1`] = `
<div>
  <div
    class="text-center"
  >
    <h2
      class="mb-2 text-2xl font-bold"
    />
    <p
      class="text-sm text-accent"
    />
  </div>
</div>
`;
